ssl/quic/quic_port.c

```diff
@@ -244,6 +251,65 @@ int ossl_quic_port_update_poll_descriptors(QUIC_PORT *port)
     return ok;
 }
 
+/*
+ * We need to determine our addressing mode. There are basically two ways we can
+ * use L4 addresses:
+ *
+ *   - Addressed mode, in which our BIO_sendmmsg calls have destination
+ *     addresses attached to them which we expect the underlying network BIO to
+ *     handle;
+ *
+ *   - Unaddressed mode, in which the BIO provided to us on the network side
+ *     neither provides us with L4 addresses nor is capable of honouring ones we
+ *     provide. We don't know where the QUIC traffic we send ends up exactly and
+ *     trust the application to know what it is doing.
+ *
+ * Addressed mode is preferred because it enables support for connection
+ * migration, multipath, etc. in the future. Addressed mode is automatically
+ * enabled if we are using e.g. BIO_s_datagram, with or without BIO_s_connect.
+ *
+ * If we are passed a BIO_s_dgram_pair (or some custom BIO) we may have to use
+ * unaddressed mode unless that BIO supports capability flags indicating it can
+ * provide and honour L4 addresses.
+ *
+ * Our strategy for determining address mode is simple: we probe the underlying
+ * network BIOs for their capabilities. If the network BIOs support what we
+ * need, we use addressed mode. Otherwise, we use unaddressed mode.
+ *
```

Comment:
I think the last paragraph lines 279 - 281 should be deleted. Perhaps other parts of comment here needs to be revisited because it came from different location. So it needs to be adjusted to match new context here.

Reply:
This doesn't seem out of place to me...

Reply:
My confusion comes from sentence found in the last paragraph:
```
 * If addressed mode is chosen, we require an initial peer address to be set. If
 * this is not set, we fail. If unaddressed mode is used, we do not require
 * this, as such an address is superfluous, though it can be set if desired.
```
`If this is not set, we fail.` -- However the function `port_update_addressing_mode()` is a void function returns nothing. So this made me wonder how we indicate the failure to caller. So this is what makes me thinking the comment needs update after being moved here.  It was just question to raise awareness.